src/hooks/WebRTC/signal.d.ts

```diff
@@ -0,0 +1,49 @@
+type Signal<Kind extends SignalKinds> = {
+    signal: Kind,
+    payload: SignalPayloadMap[Kind]
+}
+
+type SignalHandles = {
+    [kind in SignalKinds]: (signal: Signal<kind>) => void
+}
+
+type SignalKinds = "offer" | "answer" | "candidate" | "reject" | "close" | "keepalive"
+type SignalPayloadMap = {
+    "offer": RTCSessionDescriptionInit,
+    "answer": RTCSessionDescriptionInit,
+    "candidate": RTCIceCandidateInit,
+    "reject": RejectPayload,
+    "close": ClosePayload,
+    "keepalive": number,
+}
+
+type SignalCode =
```

Comment:
use enum

Reply:
I don't see the advantage of using it here, why do you suggest to use an enum?

Reply:
And the same with what you have above (SignalKinds)

Reply:
And when do you know you need an enum? You already knew that you needed an enum, you just didn't realize it, that's why you have grouped all those values ​​in a type called SignalCode. There you are hinting that all those 'numbers' are not actually numbers but rather signal codes.

Reply:
Ok, sorry for leaving this hanging and thanks for the feedback! I've seen some arguments against the use of enums in Typescript and I was skeptic to use them.

I'll make the changes on the next commit.

Reply:
Sure, there are alternatives ways to do that, but in this case, use enum it is better than a type. Why u dont like to use enums?

Reply:
It is a controversial topic in TypeScript because it adds things during run time, not during compile time. I've seen arguments both [in favour](https://dev.to/dvddpl/whats-the-problem-with-typescript-enums-2okj) and [against](https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh) their use, so I decided to use a union type before I understand better their advantages and disadvantages

Reply:
It's fine, but in 99% of use cases it's less expensive to create code that is readable and reusable than it is to lose performance. You have to think about what are the important points of your application and about the performance that using one or the other option means. (but if we're using react, I highly doubt we'd have to throttle application performance to the point of not using enums.)
 There is a well-known phrase from a Microsoft engineer who says:
"Make it correct, make it clear, make it concise, make it fast. In that order."

PD: I want to clarify that I'm not saying that you don't have to take efficiency into account, but I'm much more concerned with using sql queries in nested loops than putting some enum hahahahaha